smartlead/write.go

```diff
@@ -0,0 +1,111 @@
+package smartlead
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/spyzhov/ajson"
+)
+
+const (
+	objectNameCampaign = "campaigns"
+	objectEmailAccount = "email-accounts"
+	objectNameClient   = "client"
+)
+
+// Write method allows to
+// * create campaigns
+// * create/update email-accounts
+// * create client
+// Documentation links can be found within completeURLPath function.
+func (c *Connector) Write(
+	ctx context.Context, config common.WriteParams,
+) (*common.WriteResult, error) {
+	if len(config.ObjectName) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	url, err := c.getURL(config.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Finish URL format according to ObjectName.
+	recordIdLocation, err := completeURLPath(config, url)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.Client.Post(ctx, url.String(), config.RecordData)
+	if err != nil {
+		return nil, err
+	}
+
+	if res == nil || res.Body == nil {
+		// it is unlikely to have no payload
+		return &common.WriteResult{
+			Success: true,
+		}, nil
+	}
+
+	// write response was with payload
+	return constructWriteResult(res.Body, recordIdLocation)
+}
+
+// Maps common.WriteParams to URL format for that object.
+// In addition, returns field name where ID will be stored on successful response.
+func completeURLPath(config common.WriteParams, url *urlbuilder.URL) (string, error) {
```

Comment:
might be easier to read if we refactor this function into two create / update functions

Reply:
I reorganized it in a better way.
There is some repitition of checks if opertaion is supported for the object (it is done twice at the beginning of the method and during switch cases for Create/Update).
It has no significant impact on functionality, but should be readable